gadget/gadget.go

```diff
@@ -579,6 +582,63 @@ func compatWithPibootOrIndeterminate(m Model) bool {
 	return m == nil || m.Grade() != asserts.ModelGradeUnset
 }
 
+// Ancillary structs to sort volume structures. Each volume structure belong to
+// its own contiguousStructs unless its offset is made unknown by min-size use.
+type contiguousStructs struct {
```

Comment:
What does a contiguous structure represent in this context? Does it mean a list of ordered partitions by their offset on the volume, where a `contiguousStruct` represents a Volume?

Reply:
A contiguousStruct does not represent a Volume, a Volume would be composed of a set of contiguousStruct. Each contiguousStruct contains a first structure with a known fixed offset followed by structures for which the offset is unknown so we know for sure that they are after the first structure in contiguousStruct. I've expanded the comment to clarify this.

Reply:
Thank you for the comment clarification, that helps